redwood/src/lib.rs

```diff
@@ -131,18 +136,27 @@ fn encrypt(
 
     // For each of the recipient certificates, pull the encryption keys that
     // are compatible with by the standard policy (e.g. not SHA-1) supported by
-    // Sequoia (duh), and not revoked.
+    // Sequoia, and not revoked.
     // These filter options should be kept in sync with `Helper::decrypt()`.
     for cert in certs.iter() {
-        for key in cert
+        let keys: Vec<ValidErasedKeyAmalgamation<PublicParts>> = cert
             .keys()
             .with_policy(p, None)
             .supported()
             .alive()
             .revoked(false)
             .for_storage_encryption()
-        {
-            recipient_keys.push(key);
+            .collect();
+
+        // Each certificate must have at least one supported encryption key
+        match keys.len() {
+            0 => {
+                return Err(Error::Encryption(
+                    "At least one supported key is needed per certificate"
```

Comment:
Can we pass the cert's fingerprint here so for messages encrypted to multiple keys we can figure out which one is at fault?